joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/AstCreationForArraysTests.scala

```diff
@@ -0,0 +1,106 @@
+package io.joern.go2cpg.passes.ast
+
+import io.joern.go2cpg.testfixtures.GoCodeToCpgSuite
+import io.shiftleft.codepropertygraph.generated.{ControlStructureTypes, DispatchTypes, Operators}
+import io.shiftleft.codepropertygraph.generated.nodes.*
+import io.shiftleft.semanticcpg.language.*
+import io.shiftleft.semanticcpg.language.operatorextension.OpNodes
+import io.joern.go2cpg.testfixtures.GoCodeToCpgSuite
+
+class AstCreationForArraysTests extends GoCodeToCpgSuite {
+  "AST Creation for Array Initialization" should {
+    "be correct when a int array is initialized" in {
+      val cpg = code("""
+          |package main
+          |func main() {
+          |	a := [5]int{1,2}
+          |}
+          |""".stripMargin)
+
+      val List(assignmentCallNode, arrayInitializerCallNode) = cpg.call.l
```

Comment:
In test cases, instead of queries like `cpg.call.l` try being specific like `cpg.method("main").ast.isCall.l` so that we are creating unit test only on a specific component. I have observed in other language frontends we create extra nodes, which can come back to disturb these kind of test cases